Microsoft.HealthVault.Fhir/Codings/FhirCodesToHealthVault.cs

```diff
@@ -0,0 +1,136 @@
+    /// HV have CodableValue 
+    /// with a list of codedValue and Text
+    /// CodedValue has value, vocabularyName, family and version
+    /// HV also have a concept of vocabularyItem
+    /// with vocabularyName, version, family, value
+    /// displaytext and abbrevationtext
+    /// </summary>
+    public static class FhirCodesToHealthVault
+    {
+        public static CodableValue GetCodableValue(this CodeableConcept codeableConcept)
+        {
+            var codableValue = new CodableValue();
+            if (!string.IsNullOrEmpty(codeableConcept.Text))
+            {
+                codableValue.Text = codeableConcept.Text;
+            }
+
+            if (string.IsNullOrEmpty(codableValue.Text))
+            {
+                IEnumerable<string> displayTexts = codeableConcept.Coding.Select(coding => coding.Display).Distinct();
+                if (displayTexts.Any() && displayTexts.Count() == 1)
+                {
+                    codableValue.Text = displayTexts.Single();
+                }
+            }
+
+            foreach (var coding in codeableConcept.Coding)
+            {
+                CodedValue codedValue = GetCodedValue(coding);
+                codableValue.Add(codedValue);
+            }
+            return codableValue;
+        }
+
+        public static CodedValue GetCodedValue(this Coding coding)
+        {
+            var (value, vocabulary) = GetValueVocabularyPair(coding.Code);
+
+            string system = coding.System;
+            if (system.Equals(VocabularyUris.HealthVaultVocabulariesUri, StringComparison.OrdinalIgnoreCase))
+            {
+                system = HealthVaultVocabularies.Wc;
+            }
+
+            if (Uri.IsWellFormedUriString(system, UriKind.Absolute)
+                && string.IsNullOrEmpty(vocabulary))
+            {
+                vocabulary = HealthVaultVocabularies.Fhir;
+            }
+
+            var version = coding.Version;
+
+            var codedValue = new CodedValue();
+            if (!string.IsNullOrEmpty(value))
+            {
+                codedValue.Value = value;
+            }
+            if (!string.IsNullOrEmpty(vocabulary))
+            {
+                codedValue.VocabularyName = vocabulary;
+            }
+            if (!string.IsNullOrEmpty(system))
+            {
+                codedValue.Family = system;
+            }
+            if (!string.IsNullOrEmpty(version))
+            {
+                codedValue.Version = version;
+            }
+            return codedValue;
+        }
+
+        private static (string value, string vocabulary) GetValueVocabularyPair(string code)
+        {
+            var vocabularyValuePair = code.Split(':');
+            if (vocabularyValuePair.Count() == 2)
+                return (vocabularyValuePair[1], vocabularyValuePair[0]);
+            return (code, null);
+        }
+
+        internal static CodableValue GetRecurrenceIntervalFromPeriodUnit(Timing.UnitsOfTime? period)
```

Comment:
Let's move the recurrence interval specific items to it's own class. Could possibly make sense in Microsoft.HealthVault.Fhir/Codes/HealthVault